python/lsst/ts/observatory/control/maintel/comcam.py

```diff
@@ -240,7 +241,43 @@ async def setup_filter(self, filter):
         if filter is None:
             return
         else:
-            raise NotImplementedError("set_filter not available yet.")
+            self.rem.cccamera.cmd_setFilter.set(name=filter)
+            async with self.cmd_lock:
+                self.rem.cccamera.evt_endSetFilter.flush()
+                await self.rem.cccamera.cmd_setFilter.start(
+                    timeout=self.filter_change_timeout
+                )
+                end_setFilter = await self.rem.cccamera.evt_endSetFilter.next(
+                    flush=False, timeout=self.filter_change_timeout
+                )
+                self.log.info(f"Filter {end_setFilter.filterName} in position.")
+                return end_setFilter
```

Comment:
Can you document the return value in the docstring? You may also want to return `None` explicitly when filter is `None`, since you do have a return value.

Reply:
Done.